app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt

```diff
@@ -0,0 +1,311 @@
+            PRECISION_6
+          )
+        )
+      }
+
+      override fun onFailure(call: Call<DirectionsResponse>, t: Throwable) {}
+    })
+  }
+
+  /**
+   * Adds a GeoJsonSource to a style to visualize a LineString
+   */
+  private fun addSourceForRoute(style: Style, lineString: LineString) {
+    style.addSource(
+      geoJsonSource(GEOJSON_SOURCE_ID) {
+        geometry(lineString)
+      }
+    )
+  }
+
+  /**
+   * Animate the route using FreeCamera API
+   */
+  private fun animateRoute(lineString: LineString) {
+    // get the overall distance of each route so we can interpolate along them
+    val routeDistance = TurfMeasurement.length(
+      lineString, TurfConstants.UNIT_KILOMETERS
+    )
+
+    // get reference to the location layer
+    val locationLayer = mapboxMap.getStyle()!!.getLayerAs<LocationIndicatorLayer>(
+      LOCATION_LAYER_ID
+    )
+
+    // cache the camera
+    val camera = mapboxMap.getFreeCameraOptions()
+
+    // get a reference to elevations of the animation
+    // note: to call getElevation the coordinate needs to be in the viewport
+    val startElevation = mapboxMap.getElevation(POINT_START)!!.toDouble()
+    val endElevation = mapboxMap.getElevation(POINT_END)!!.toDouble()
+    val minElevation = minOf(startElevation, endElevation)
+    val maxElevation = maxOf(startElevation, endElevation)
+
+    // use time animator to animate the lineString and location layer
+    val timeAnimator = TimeAnimator()
+    timeAnimator.setTimeListener { animator, totalTime, _ ->
+
+      // phase determines how far through the animation we are
+      val phase: Double = totalTime / ANIMATION_DURATION
+
+      // phase is normalized between 0 and 1
+      // when the animation is finished, cancel the animation
+      if (phase > 1) {
+        animator.cancel()
+      }
+
+      // use phase to get a point that is the appropriate distance along the route
+      val cameraLookingAt = TurfMeasurement.along(
+        lineString, routeDistance * phase,
+        TurfConstants.UNIT_KILOMETERS
+      )
+
+      // at start, we hover the current position first before trailing it
+      var cameraPhase = phase - PHASE_DROP_OFF
+      if (cameraPhase < 0) {
+        cameraPhase = 0.0
+      }
+
+      // use phase that trails behind to be able to create a tilted camera
+      val cameraLocation = TurfMeasurement.along(
+        lineString, routeDistance * cameraPhase,
+        TurfConstants.UNIT_KILOMETERS
+      )
+
+      // calculate the elevation of the user location
+      val elevation = minOf(maxElevation, (maxElevation * phase) + minElevation)
+
+      // Update location indicator
+      locationLayer.location(
+        listOf(
+          cameraLookingAt.latitude(),
+          cameraLookingAt.longitude(),
+          elevation
+        )
+      )
+
+      // calculate an elevation difference between the user position and the camera
+      val elevationDifference = BASE_CAMERA_ELEVATION + elevation
```

Comment:
Is it indeed difference? Looks more like a sum to me 😄

Reply:
updated the comments